<a href="https://colab.research.google.com/github/ahtisham-maliik/ML-Exercise/blob/main/ML%20KFold%20Cross%20Validation%2002/ML_KFold_Cross_Validation_02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [39]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [40]:
iris = load_iris()

**Logistic Regression**

In [41]:
l_scores = cross_val_score(LogisticRegression(), iris.data, iris.target)
l_scores

/usr/local/lib/python3.10/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [42]:
np.average(l_scores)

0.9733333333333334

**Decision Tree**

In [43]:
d_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
d_scores

array([0.96666667, 0.96666667, 0.9       , 1.        , 1.        ])

In [44]:
np.average(d_scores)

0.9666666666666668

**Support Vector Machine (SVM)**

In [45]:
s_scores = cross_val_score(SVC(), iris.data, iris.target)
s_scores

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [46]:
np.average(s_scores)

0.9666666666666666

**Random Forest**

In [47]:
r_scores = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target)
r_scores

array([0.96666667, 0.96666667, 0.93333333, 0.96666667, 1.        ])

In [48]:
np.average(r_scores)

0.9666666666666668

**Best score so far is from Logistic Regression 0.973**